assets/js/nebula.js

```diff
@@ -2728,40 +2728,57 @@ nebula.cf7LocalStorage = function(){
 		return false;
 	}
 
-	jQuery('.wpcf7-textarea, .wpcf7-text').each(function(){
+	jQuery('.wpcf7-textarea, .wpcf7-text, .wpcf7-select, .wpcf7-checkbox input, .wpcf7-radio input').each(function(){
 		var thisLocalStorageVal = localStorage.getItem('cf7_' + jQuery(this).attr('name'));
-
-		//Fill textareas with localstorage data on load
+		//Complete inputs with localstorage data on load
 		if ( !jQuery(this).hasClass('do-not-store') && !jQuery(this).hasClass('.wpcf7-captchar') && thisLocalStorageVal && thisLocalStorageVal !== 'undefined' && thisLocalStorageVal !== '' ){
-			if ( jQuery(this).val() === '' ){ //Don't overwrite a field that already has text in it!
-				jQuery(this).val(thisLocalStorageVal).trigger('keyup');
+			if ( thisLocalStorageVal !== null ){
+				if ( jQuery(this).attr('type') === 'checkbox' || jQuery(this).attr('type') === 'radio' ){
+					jQuery(this).prop('checked', (jQuery(this).val() === thisLocalStorageVal));
+				} else {
```

Comment:
I'd *love* if we could find a way to not need this else... it seems tantalizingly close, but it may be necessary.